### ISSUE TITLE
Follow up - deploy locally 

### ISSUE THREAD

**Issue by JeffKatzy** [regisb/tutor#135]
Thanks for all of the help @regisb.  I was able to make it past that issue, but now am running into another one that does not allow me to deploy locally while linking my code.

## Steps to reproduce
```
env | grep EDX 
EDX_PLATFORM_SETTINGS=tutor.development
EDX_PLATFORM_PATH=/Users/jeff/Documents/jigsaw/software/edx-software/downloaded-edx/edx-platform
```

```
make lms-runserver

docker-compose run -e SETTINGS=tutor.development --volume="$(pwd)/../../build/openedx/themes:/openedx/themes" -e USERID=$(id -u) --volume="/Users/jeff/Documents/jigsaw/software/edx-software/downloaded-edx/edx-platform:/openedx/edx-platform" -p 8000:8000 lms ./manage.py lms runserver 0.0.0.0:8000
Creating local_mongodb_1       ... done
Creating local_smtp_1          ... done
Creating local_rabbitmq_1      ... done
Creating local_elasticsearch_1 ... done
Creating local_mysql_1         ... done
Creating local_memcached_1     ... done
Creating local_forum_1         ... done
creating new user 'openedx' with UID 501
chown: changing ownership of '/openedx/edx-platform/common/static/edx-ui-toolkit/js': No such file or directory
chown: changing ownership of '/openedx/edx-platform/common/static/edx-pattern-library/js': No such file or directory
chown: changing ownership of '/openedx/edx-platform/common/static/edx-pattern-library/fonts': No such file or directory
make: *** [lms-runserver] Error 1
````

I receive the same error message when running `make cms-runserver`

## Things I tried

1. Neither `docker-compose logs` nor `docker container logs -f local_lms_run_9bc186300ca6` provided me with more info.  
2. On one of the launches, saw an error of `Invalid http_host header` so led me to copy over the settings, also as per the documentation 

```
cp -r tutor/deploy/templates/openedx/settings/cms/ downloaded-edx/edx-platform/cms/envs/tutor/
cp -r tutor/deploy/templates/openedx/settings/lms/ downloaded-edx/edx-platform/lms/envs/tutor/
``` 

* Notice that my path to the settings is different than that in the documentation of `config/openedx/tutor/lms/`, I couldn't find files in that path.

## Additional info (IMPORTANT)

```

[tutor (master)*]$ make info
uname -a
Darwin Jeffreys-MacBook-Air.local 18.2.0 Darwin Kernel Version 18.2.0: Mon Nov 12 20:24:46 PST 2018; root:xnu-4903.231.4~2/RELEASE_X86_64 x86_64
-------------------------
git rev-parse HEAD
c9314e196d66c1275d936e20dc44a65510506962
-------------------------
docker version
Client: Docker Engine - Community
 Version:           18.09.1
 API version:       1.39
 Go version:        go1.10.6
 Git commit:        4c52b90
 Built:             Wed Jan  9 19:33:12 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.1
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       4c52b90
  Built:            Wed Jan  9 19:41:49 2019
  OS/Arch:          linux/amd64
  Experimental:     false
-------------------------
docker-compose --version
docker-compose version 1.23.2, build 1110ad01
-------------------------
echo $EDX_PLATFORM_PATH
/Users/jeff/Documents/jigsaw/software/edx-software/downloaded-edx/edx-platform
echo $EDX_PLATFORM_SETTINGS
tutor.development

```

**Comment by regisb** [comment]
The big problem, when you mount a local edx-platform folder inside a container, is that there is a lot of stuff that needs to be done inside this repo to make it work. For instance, the `edx-platform/common/static/edx-ui-toolkit/js` folder is a symlink to `edx-platform/node_modules/edx-ui-toolkit/src/js`, which only exists if nodejs modules have been installed inside the edx-platform/node_modules folder.

Long story short, you should run:

```
make lms # login the lms
npm install # install nodejs requirements
```

Then, the `make lms-runserver` commands should work.

**Comment by JeffKatzy** [comment]
Hey @regisb, thanks again.  Ah ok cool so I made some progress by first running `npm install` on the host's version of edx, and then running `make lms`.   

I've gotten stuck at the following error:

`chown: changing ownership of '/openedx/edx-platform/node_modules/.bin/har-validator': No such file or directory`

Even though after running `npm install --only=production --save @har/cli` and `npm install har-validator`, I able to find `/.bin/har-validator` my host's version of edx, it seems like it's not able to be referenced inside the container?  

Here is the error in full: 
```

make lms
docker-compose run --rm -e SETTINGS=tutor.development --volume="$(pwd)/../../build/openedx/themes:/openedx/themes" -e USERID=$(id -u) --volume="/Users/jeff/Documents/jigsaw/software/edx-software/downloaded-edx/edx-platform:/openedx/edx-platform" -p 8000:8000 lms bash
Starting local_elasticsearch_1 ... done
Starting local_smtp_1          ... done
Starting local_memcached_1     ... done
Starting local_rabbitmq_1      ... done
Starting local_mysql_1         ... done
Starting local_mongodb_1       ... done
Starting local_forum_1         ... done
creating new user 'openedx' with UID 501
chown: changing ownership of '/openedx/edx-platform/node_modules/.bin/har-validator': No such file or directory
make: *** [lms] Error 1

```

I was going to try to add an `npm install` to the docker file, but it seems like this already occurs [in the dockerfile](https://github.com/regisb/tutor/blob/c42b747e3538eb8f0ead1534c381933afdc0b87e/build/openedx/Dockerfile#L51) . Any solution/path forward you recommend?

**Comment by regisb** [comment]
I don't understand why this error occurs, but you should run `npm install` inside the docker container to make sure you are using the right version of npm/nodejs.

Adding `npm install` would not work becauseyour edx-platform repo would not be mounted inside the container. Mounting volumes happens at runtime, not build time.

**Comment by JeffKatzy** [comment]
ok! well after 4 hours of trying it my way, I decided to downgrade the npm version from 11+ down to 8.9.3.  That got me pretty far.  I also removed the reference to Anaconda in my `.bash_profile` which is where my node version was installed.  

currently running into this  `app_label` error `Model class openedx.core.djangoapps.bookmarks.models.Bookmark doesn't declare an explicit app_label`, but that seems to be a fixable Edx thing.  Thanks for all the help!!
